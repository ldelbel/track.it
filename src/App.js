import Navbar from "./common/components/Navbar";
import History from "./pages/history";
import Menu from "./common/components/Menu";

function App() {
  return (
    <div className="App">
      <Navbar />
      <History />
      <Menu />
    </div>
  );
}

export default App;
