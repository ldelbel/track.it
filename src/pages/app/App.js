import {
  BrowserRouter as Router,
  Switch,
  Route,
  useRouteMatch
} from "react-router-dom";
import { Provider } from "react-redux";
import Navbar from "../../common/components/Navbar";
import History from "./history";
import Progress from "./progress";
import New from "./new";
import Menu from "../../common/components/Menu";
import store from "../../store/store";


function App() {
  let { path } = useRouteMatch();

  return (
    <Provider store={store}>
      <div className="App">
        <Navbar />
          <Switch>
            <Route exact path={path} component={New} />
            <Route path={`${path}/history`}  component={History} />
            <Route path={`${path}/progress`} component={Progress} />
          </Switch>
        <Menu />
      </div>
    </Provider>
  );
}

export default App;
