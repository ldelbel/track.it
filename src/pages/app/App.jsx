import { useState, useEffect } from 'react';
import {
  Switch,
  Route,
  useRouteMatch,
  useHistory
} from "react-router-dom";
import { connect } from 'react-redux';
import Navbar from "../../common/components/Navbar";
import History from "./history";
import Progress from "./progress";
import New from "./new";
import Menu from "../../common/components/Menu";
import { addRunningSession, fillList } from '../../actions';
import { userID, fetchUserData } from '../../api';
import Loading from 'react-loading-components';


const App = props => {
  const { addRunningSession, username, fillList, runningSessions } = props;
  let { path } = useRouteMatch();
  const [isLoading, setIsLoading] = useState(true);
  let history = useHistory();

  const loadFromBackend = async () => {
    const id = await userID(username);  
    const data = await fetchUserData(id);
    fillList(id,data);
    localStorage.setItem('id', JSON.stringify(id));
    localStorage.setItem('runningSessions', JSON.stringify(data));
    setIsLoading(false);
  }

  const loadFromLocalStorage = () => {
    const id = JSON.parse(localStorage.getItem('id'));
    const data = JSON.parse(localStorage.getItem('runningSessions'));
    fillList(id,data);
    setIsLoading(false);
  }

  useEffect(() => {
    if(!localStorage['user']){
      history.push('/');
    } else {
      if(localStorage['runningSessions']){
        loadFromLocalStorage();
      } else {
        loadFromBackend();
      }
    }    
  }, [])

  return (
    isLoading ?
      ( <main style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', background: 'var(--blue)', height: '100vh'}}>
          <Loading type='ball_triangle' width={100} height={100} fill='#fff' />
        </main>
      ):(
      <div className="App">
      <Navbar />
      <Switch>
        <Route 
        exact path={path}
        render={props => <New {...props} addRunningSession={addRunningSession} /> }
        />
        <Route 
        path={`${path}/history`}
        render={props => <History {...props} runningSessions={runningSessions} /> }
        />
        <Route
        path={`${path}/progress`}
        render={props => <Progress {...props} runningSessions={runningSessions} /> }
        />
      </Switch>
      <Menu />
    </div>
    )
  );
}

export default connect(state => state, { addRunningSession, fillList })(App);

