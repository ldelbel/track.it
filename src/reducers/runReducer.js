import { ADD_RUNNING_SESSION, SET_USER, FILL_LIST } from '../actions';

let initialState = {
  id: 0,
  username: '',
  runningSessions: [],
};

if (localStorage.user) {
  initialState = {
    id: localStorage.id,
    username: localStorage.user,
    runningSessions: JSON.parse(localStorage.runningSessions),
  };
}

const runReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_RUNNING_SESSION:
      return {
        ...state,
        runningSessions: state.runningSessions.concat(action.payload),
      };
    case SET_USER:
      return {
        ...state,
        username: action.payload,
      };
    case FILL_LIST:
      return {
        ...state,
        id: action.payload[0],
        runningSessions: action.payload[1],
      };
    default:
      return state;
  }
};

export default runReducer;
