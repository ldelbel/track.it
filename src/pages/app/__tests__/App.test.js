import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import App from '../App';
import runReducer from '../../../reducers/runReducer';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useRouteMatch: () => ({ url: '/app' }),
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('app page', () => {
  const store = createStore(runReducer);

  it('renders correctly', () => {
    const app = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
