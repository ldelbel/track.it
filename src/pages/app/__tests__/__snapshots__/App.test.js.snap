// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app page renders correctly 1`] = `
<main
  style={
    Object {
      "alignItems": "center",
      "background": "var(--blue)",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
    }
  }
>
  <svg
    height={100}
    stroke="#fff"
    viewBox="0 0 57 57"
    width={100}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <g
        strokeWidth="2"
        transform="translate(1 1)"
      >
        <circle
          cx="5"
          cy="50"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="50;5;50;50"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="5;27;49;5"
          />
        </circle>
        <circle
          cx="27"
          cy="5"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="5"
            repeatCount="indefinite"
            to="5"
            values="5;50;50;5"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="27"
            repeatCount="indefinite"
            to="27"
            values="27;49;5;27"
          />
        </circle>
        <circle
          cx="49"
          cy="50"
          r="5"
        >
          <animate
            attributeName="cy"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            repeatCount="indefinite"
            values="50;50;5;50"
          />
          <animate
            attributeName="cx"
            begin="0s"
            calcMode="linear"
            dur="2.2s"
            from="49"
            repeatCount="indefinite"
            to="49"
            values="49;5;27;49"
          />
        </circle>
      </g>
    </g>
  </svg>
</main>
`;
